src/pages/Product/Product.js

```diff
@@ -0,0 +1,248 @@
+import { Link } from "react-router-dom";
+import { useParams } from "react-router";
+import { useCart } from "../../contexts/CartProvider";
+import { useProducts } from "../../contexts/ProductsProvider";
+import { useWishlist } from "../../contexts/WishlistProvider";
+import styles from "./Product.module.css";
+import { BiCheckDouble } from "react-icons/bi";
+import DefaultWithoutSearch from "../../layouts/DefaultWithoutSearch";
+import {
+  FaRegHeart,
+  FaHeart,
+  FaStar,
+  FaCartPlus,
+  FaShoppingCart,
+} from "react-icons/fa";
+import { ReactComponent as EmptyProductSvg } from "./EmptyProductImage.svg";
+
+const Product = () => {
+  const { id } = useParams();
+  const { products, isProductsLoading } = useProducts();
+  const { isCartLoading, cartState, addToCart } = useCart();
+  const {
+    isWishlistLoading,
+    wishlistState,
+    removeFromWishlist,
+    addToWishlist,
+  } = useWishlist();
+  const product = products.find((product) => product._id === id);
+
+  const handleAddToCart = async (product, productId) => {
+    await addToCart(product, productId);
+  };
+
+  const handleAddOrRemoveWishListItems = async (productId) => {
+    if (wishlistState.some((item) => item.product._id === productId)) {
+      await removeFromWishlist(productId);
+    } else {
+      await addToWishlist(productId);
+    }
+  };
+
+  if (isProductsLoading) {
+    return (
+      <DefaultWithoutSearch>
+        <h1 className="overlay">Loading ...</h1>
```

Comment:
Try not to add Fullscreen overlay loaders everywhere.
First-time loading is fine, but for route changing, it's a bad UX. Use page / component specific loaders.